### ISSUE TITLE
Cast exception when getting metadata

### ISSUE THREAD

**Issue by KavinduZoysa** [ballerina-platform/ballerina-lang#24367]
**Description:**
source code: 
```
public main() 
    int xxx;
    Foo f  new(3, "John");
}
```

error:
```
java.lang.ClassCastException: io.ballerinalang.compiler.internal.syntax.ExternalTreeNodeList cannot be cast to io.ballerinalang.compiler.syntax.tree.MetadataNode
	at io.ballerinalang.compiler.syntax.tree.ModuleVariableDeclarationNode.metadata(ModuleVariableDeclarationNode.java:37)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:490)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:405)
	at io.ballerinalang.compiler.syntax.tree.ModuleVariableDeclarationNode.apply(ModuleVariableDeclarationNode.java:67)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:472)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:405)
	at io.ballerinalang.compiler.syntax.tree.ModulePartNode.apply(ModulePartNode.java:54)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.accept(BLangNodeTransformer.java:430)
	at org.wso2.ballerinalang.compiler.parser.Parser.generateCompilationUnitNew(Parser.java:143)
	at org.wso2.ballerinalang.compiler.parser.Parser.parseNew(Parser.java:116)
	at org.wso2.ballerinalang.compiler.PackageLoader.parse(PackageLoader.java:500)
	at org.wso2.ballerinalang.compiler.PackageLoader.loadEntryPackage(PackageLoader.java:380)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:156)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackage(Compiler.java:174)
	at org.wso2.ballerinalang.compiler.Compiler.compile(Compiler.java:91)
	at org.wso2.ballerinalang.compiler.Compiler.build(Compiler.java:99)
	at org.ballerinalang.packerina.task.CompileTask.execute(CompileTask.java:52)
	at org.ballerinalang.packerina.TaskExecutor.executeTasks(TaskExecutor.java:38)
	at org.ballerinalang.packerina.cmd.RunCommand.execute(RunCommand.java:310)
	at java.util.Optional.ifPresent(Optional.java:159)
	at org.ballerinalang.tool.Main.main(Main.java:57)
```

This source code is parsed as 3 module var declarations. Getting above cast exception for third declaration `Foo f`

**Comment by SupunS** [comment]
This recovers and ands with a module level var decl. Then fails at the transformer due to #24878

**Comment by rdulmina** [comment]
Fixed in #26949
